### ISSUE TITLE
Redirection operator `>` should not be hardcoded to files

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#23794]
### Summary of the new feature / enhancement

Currently, `>` will internally call `Out-File`.  Instead, it should probably call `Out-Provider` (or something like that) so that any provider can participate: `1 > variable:a`.

### Proposed technical implementation details (optional)

If a provider path isn't provided, default to filesystem

**Comment by rhubarb-geek-nz** [comment]
One can simply pipe through Set-Variable if you want that

```
PS>  'foo' | Set-Variable 'bar'
PS> echo $bar
foo
```

No?

**Comment by SteveL-MSFT** [comment]
That works for output (or STDOUT), but we don't have `2|` for error stream so you have to use `2>` which only writes to files currently.  Although we have https://github.com/PowerShell/PowerShell/pull/20381 which would allow redirection to a variable and during that code review I thought of this.

**Comment by rhubarb-geek-nz** [comment]
> so you have to use `2>` which only writes to files currently

If you use `*>&1` you can filter for `ErrorRecord` in the success output. [Example](https://github.com/rhubarb-geek-nz/PipelineBroadcast/blob/main/test.ps1).

**Comment by SteveL-MSFT** [comment]
> > so you have to use `2>` which only writes to files currently
> 
> If you use `*>&1` you can filter for `ErrorRecord` in the success output. [Example](https://github.com/rhubarb-geek-nz/PipelineBroadcast/blob/main/test.ps1).

Yes, but sometimes you want `foo 2>variable:stderr >variable:stdout`

There's also https://github.com/PowerShell/PowerShell/pull/21565 for those that want `tee-object` for errors

**Comment by jborean93** [comment]
I agree this would be great to have, I currently use the following to capture stdout/stderr as separate vars

```powershell
$stdout = $null
$stderr = & { my.exe | Set-Variable -Name stdout } 2>&1 | ForEach-Object ToString
```

The `ForEach-Object ToString` is only needed for WinPS to capture stderr as a string and not the ErrorRecords in the stream. Having a way to do it directly with redirection would be really nice as I can avoid the above complexity.

**Comment by mklement0** [comment]
@jborean93, the more succinct cross-edition formulation of your code is:

```powershell
$stdout, [string[]] $stderr = (my.exe 2>&1).Where({ $_ -is [string] }, 'Split')
```

What #20381 now enables:

```powershell
my.exe 1> variable:stdout 2> variable:stderr
```

Alternatively:

```powershell
$stdout = my.exe 2> variable:stderr
```

That is, #20381 is sufficient for targeting _variables_ via redirections, via the `variable:` drive.

The issue at hand is about  _generalizing_ this approach, allowing _any provider's items_ as the target.
